Funcionalidad del carrito de compras

RonaldoSwift opened the pull request:
### Porque? (Motivación: Cual es la razón para este cambio)

* El usuario desea ver el carrito de compras

### Que? (Contexto: Como se comportaba el código antes del cambio)

* Antes no tenia un carrito de compras 

### Como? (Implementación: Que cambios se hicieron)

* Agregue una imagen de tacho y nueva lista de Comidas View


### Evidencia (Screenshots/Video/Logs, etc.)

*<img width="297" alt="Screen Shot 2021-12-10 at 11 58 07" src="https://user-images.githubusercontent.com/61673331/145612220-c1a6dbac-6b2c-4ffa-a8e7-eafabf76dcba.png">


### Checklist:
[ ] He creado/actualizado la documentación necesaria.
[ ] He agregado pruebas unitarias para cubrir mis cambios.
[  X ] Todas las pruebas unitarias existentes siguen funcionando.

RonaldoSwift committed:
ok

CarlitosDroid commented on POS/ListaComidaView.swift:
```diff
@@ -0,0 +1,58 @@
+    
+    var body: some View {
+        
+        ForEach(comidas, id: \.id){ comida in
```
No olvides usar el `ForEach` dentro de un padre, puedes usar un `List`, por ejemplo:

```
List {
    ForEach {}
}
```

RonaldoSwift replied:
Hecho, agregue el list

CarlitosDroid commented on POS/ListaComidaView.swift:
```diff
@@ -0,0 +1,58 @@
+        
+        ForEach(comidas, id: \.id){ comida in
+            ZStack{
+                Image("Fondo")
```
Deberias quitar el `ZStack` y `Image("Fondo")`. Simplemente le pones un `background` con el color que deseas a tu `HStack`

RonaldoSwift replied:
Hecho

CarlitosDroid commented on POS/ListaComidaView.swift:
```diff
@@ -0,0 +1,58 @@
+                                Text("$\(comida.precio)")
+                                    .foregroundColor(Color.gray)
+                            }
+                            Image("Numero2")
```
No uses Imagenes, porque sino no estaras aprovechando el potencial de SwiftUI. Esto lo puedes hacer algo si:

```
Text("20")
    .padding()
    .background(Color.red)
    .border(Color.green, width: 4)
    .cornerRadius(8)                           
```

RonaldoSwift replied:
Hecho,agregue el Text enves de la imagen

CarlitosDroid commented on POS/ListaComidaView.swift:
```diff
@@ -0,0 +1,58 @@
+                                .resizable()
+                                .frame(width: 60, height: 60, alignment: .trailing)
+                        }
+                        Image("Imagen2")
```
Lo mismo aquí, no uses imagenes, hazlo con puro SwiftUI

RonaldoSwift replied:
Hecho,convertido de Imagen a codigo  SwiftUI

CarlitosDroid commented on POS/ListaComidaView.swift:
```diff
@@ -0,0 +1,58 @@
+
+import SwiftUI
+
+struct ListaComidaView: View {
```
El nombre de tu estructura nos puede confundir porque ya tenemos otra pantalla donde se muestra una lista de comida. Esta vista es mas un carrito de compras y podrías llamarlo `CarritoView`.

RonaldoSwift replied:
Hecho, cambiado por CarritoView

RonaldoSwift committed:
- Cambio de imagen a  Text en código SwiftUI
- Agregue  nueva propiedad "cantidad" en Estructura Comida
-Cambie nombre de estructura ListaComidaView a CarritoView
-Agreue List en el ForEach
- En Preview agregue el .prederredColorScheme(.dark) para modo oscuro

CarlitosDroid reviewed (approved):
'APPROVE' performed by Xcode.

RonaldoSwift committed:
Merge branch 'main' into E2
